WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs

```diff
@@ -6,16 +6,101 @@
 using ComparisonEngineUI.ViewModels;
 using Xamarin.Forms;
 using Xamarin.Forms.Xaml;
+using ComparisonEngineUI.Models;
+using System.Text.RegularExpressions;
+using ComparisonEngineUI.Data;
 
 namespace ComparisonEngineUI.Views
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class EditDrinkPricePage : ContentPage
     {
+        private List<Bar> _barList { get; set; }
+        public List<Bar> BarList
+        {
+            get
+            {
+                return _barList;
+            }
+            set
+            {
+                if (value != _barList)
+                {
+                    _barList = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+        private List<Drink> _drinkList { get; set; }
+        public List<Drink> DrinkList
+        {
+            get
+            {
+                return _drinkList;
+            }
+            set
+            {
+                if (value != _drinkList)
+                {
+                    _drinkList = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+        private List<Drink> initialDrinkList { get; set; }
+        private List<Bar> initialBarList { get; set; }
         public EditDrinkPricePage()
         {
             InitializeComponent();
             this.BindingContext = new EditDrinkPriceViewModel();
+            var restService = new RestService();
+            initialBarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+            initialDrinkList = Task.Run(async () => await restService.GetData<List<Drink>>(Constants.DrinksUrl)).Result;
+        }
+
+        private async void OnSaveClicked(object sender, EventArgs e)
+        {
+
+            if (Regex.IsMatch(newEntryPrice.Text, @"(^[1-9]\d*(.\d{1,2})?$)|(^0(\.\d{1,2})?$)"))
```

Comment:
regex probably should have been moved to a different class

Reply:
To different class maybe, but not further(I think we were discussing about moving regex to backend), as I understand regex, it's more of a front end thing, because it doesn't allow me to enter anything but it's set format, so having that in backend could probably slow down the entering.